app/controllers/schools/calendars_controller.rb

```diff
@@ -25,12 +26,15 @@ def create
         .where(id: calendar_params[:cohort_ids])
         .pluck(:id)
 
-      @course.calendars.create!(calendar_params)
-
-      authorize(@calendar, policy_class: Schools::CalendarEventPolicy)
+      authorize(@course, policy_class: Schools::CalendarEventPolicy)
+      calendar = @course.calendars.create!(calendar_params)
 
       flash[:success] = I18n.t("calendars.create.success")
-      redirect_to school_course_calendar_events_path(@course)
+
+      redirect_to school_course_calendar_events_path(
```

Comment:
I have a feeling this redirect after creation of the calendar wouldn't be necessary, as you will always have no events immediately after creation.

Reply:
@mahesh-krishnakumar 
I think, redirection to the previously calendar would be nice?

Reply:
I think in general, yes. But the only difference I see in this particular action is that you are filtering the page using a newly created calendar, which obviously doesn't have any events. This filter would only be useful if the immediate next step is creating a new event and the new calendar is preselected. But immediately after calendar creation, seeing the empty events page suddenly from the previous one filled with events would surprise the user is personally what I feel.